src/scripts/utils/batch.js

```diff
@@ -0,0 +1,22 @@
+// batchExecute(cb, 2, [1,2,3,4,5,6,7]) will return
+// [cb([1,2]), cb([3,4]), cb([5,6]), cb([7])]
+
+// batchExecute(cb, 2, [1,2,3,4,5,6,7], ['a','b','c','d','e','f','g']) will return
+// [cb([1,2], ['a','b']), cb([3,4], ['c','d']), cb([5,6], ['e','f']), cb([7], ['g'])]
+
+const batchExecute = (cb, batchSize = Infinity, ...argsArrays) => {
+  let results = []
+
+  for (let i = 0, len = Math.max(...argsArrays.map(a => a.length)); i< len; i+=batchSize) {
+      const args = argsArrays.map(a => a.slice(i, i + batchSize))
+      try {
+          const resultSlice = cb(...args)
```

Comment:
But this calls right away, right? 
In this example we want to do it in blocking batches. Otherwise we send all the TX at the same time, and I'm not sure if that is a good idea. 

Maybe is allowing cb to return a promise, and waiting for the promise to complete.
For the token approvals it was in blocking batches. If we are not sure what's best, we can get fancy and allow a an extra param with the maximum concurrency N (so it'll allow to execute N cb, but I'll wait for the N+1)

Reply:
Done